src/components/Modals/BuySell/providers/GemManager.tsx

```diff
@@ -0,0 +1,191 @@
+import { ArrowBackIcon, CheckIcon, ChevronRightIcon, CopyIcon, ViewIcon } from '@chakra-ui/icons'
+import {
+  Box,
+  Button,
+  Flex,
+  IconButton,
+  Input,
+  InputGroup,
+  InputRightElement,
+  Stack,
+  Text as RawText,
+  useToast
+} from '@chakra-ui/react'
+import { getConfig } from 'config'
+import queryString from 'querystring'
+import { useEffect, useMemo, useState } from 'react'
+import { useTranslate } from 'react-polyglot'
+import { AssetIcon } from 'components/AssetIcon'
+import { SlideTransition } from 'components/SlideTransition'
+import { Text } from 'components/Text'
+import { useModal } from 'context/ModalProvider/ModalProvider'
+import { useWallet } from 'context/WalletProvider/WalletProvider'
+
+import { BuySellAction, BuySellActionType, BuySellAsset } from '../BuySell'
+import { AssetSearch } from '../components/AssetSearch/AssetSearch'
+import { getAssetLogoUrl } from '../components/AssetSearch/helpers/getAssetLogoUrl'
+import { BuySellActionButtons } from '../components/BuySellActionButtons'
+
+const middleEllipsis = (address: string, cut: number) =>
+  `${address.slice(0, cut)}...${address.slice(-1 * cut)}`
+
+const GEM_ENV = getConfig().REACT_APP_GEM_ENV
+const GEM_API_KEY = getConfig().REACT_APP_GEM_API_KEY
+const GEM_URL = getConfig().REACT_APP_GEM_URL
+
+export const GemManager = () => {
+  const translate = useTranslate()
+  const toast = useToast()
+  const { buysell } = useModal()
+
+  const [asset, setAsset] = useState<BuySellAsset | null>()
+  const [isSelectingAsset, setIsSelectingAsset] = useState(false)
+  const [verified, setVerified] = useState<boolean | null>(null)
```

Comment:
Do we actually use `verified` anywhere apart from button state/styling? From the ACC, this should allow to verify the address on a hardware wallet